### PR TITLE
feat(core): add observational bashkit hooks for virtual_bash

### PR DESCRIPTION
**Pull request by chaliy**
## What

Installs observational-only `bashkit` interceptor hooks on every `Bash` instance built by the `virtual_bash` capability. The hooks emit structured `tracing` events tagged with the active `session_id` so operators can audit per-builtin activity and interpreter errors without touching the script text or stdout.

Hooks registered:
- `before_tool` — logs `tool`, `arg_count` (values not logged)
- `after_tool` — logs `tool`, `exit_code`, `stdout_bytes` (content not logged)
- `on_error` — logs a 256-byte UTF-8-safe truncated preview of the error message at `warn!`

HTTP hooks (`before_http` / `after_http`) are deliberately not installed: `virtual_bash` does not enable bashkit's `http_client` feature and has no network builtins (TM-BASH-003).

## Why

EVE-299 — bashkit v0.1.18 ships a hook system, but `virtual_bash` still builds Bash instances without any interceptors. The existing `tracing::info!` at the end of `execute()` only exposes aggregate counts from `TraceEventKind` events. Per-builtin audit granularity (which builtin ran, with what exit code) and first-class interpreter-error capture were not previously reachable without additional bashkit features. The issue asks for a narrow observational prototype first; this is that prototype.

## How

- Added `install_observability_hooks(builder, session_id) -> BashBuilder` in `crates/core/src/capabilities/virtual_bash.rs`. Every hook returns `HookAction::Continue` — zero behavior change.
- Wired the installer into both `Tool::execute` and `BackgroundExecutableTool::execute_background` for `BashTool`, so both the foreground and background bash paths get the same telemetry.
- Added `redact_for_log(msg, max_bytes)` helper for bounded UTF-8-safe error previews.
- Added four unit tests: three cover `redact_for_log` edge cases (short, truncated + marker, multi-byte char boundary); one end-to-end smoke test composes the installer with an observer hook, runs a script, and asserts both the exit code and that `after_tool` fired.
- Documented the hook surface in `specs/bashkit-requirements.md` (new "Observability Hooks" section).

## Risk

- Low.
- Every hook is observational; closures return `HookAction::Continue` and cannot widen bashkit's existing limits (TM-BASH-005/006/007) or sandbox boundaries. The hook pipeline is a `Vec::is_empty()` check when no hooks are registered and a few closure calls otherwise — no perceptible overhead. Worst case is a noisy `tracing` subscriber, which is configurable per-deployment.

## Security

- Threat categories reviewed: **TM-BASH** (Bash sandbox), **TM-WEB** (HTTP hooks), general logging exposure.
- Findings and resolutions:
  - **Secret exfiltration via logs.** Hook callbacks log only structural metadata (tool name, arg count, exit code, byte lengths). Argument values and builtin stdout are never logged — those surfaces may carry tenant paths, URLs, or embedded secrets.
  - **Unbounded error payloads.** `on_error` can receive arbitrarily long error messages. Each error is truncated to 256 bytes on a UTF-8 char boundary before being passed to `tracing::warn!`, preventing log amplification.
  - **Sandbox bypass.** Every hook returns `HookAction::Continue`. None modify `ExecInput`, `ToolEvent`, or cancel execution. Bashkit's limits, allowlist, and VFS adapter boundaries remain the sole enforcement surface.
  - **Network allowlist bypass (TM-BASH-003).** HTTP hooks are not installed; they would require bashkit's `http_client` feature which this crate does not enable. No network builtins exist inside `virtual_bash`, so there is no TM-WEB-relevant surface to observe. Comment on the helper documents this so a future contributor enabling `http_client` registers HTTP hooks explicitly.
- No new THREAT comments added; no existing THREAT comments in `virtual_bash.rs` were moved or altered.

## Checklist
- [x] Tests added or updated (4 new unit tests in `crates/core/src/capabilities/virtual_bash.rs`)
- [x] Backward compatibility considered (observational-only, no output / behavior changes)
- [x] Security review performed against relevant threat model categories (TM-BASH, TM-WEB)
- [ ] All review comments addressed (code change or written reasoning)

### CODE REVIEW

File: crates/core/src/capabilities/virtual_bash.rs [hunk 1]
```diff
@@ -577,6 +579,71 @@ impl BackgroundExecutableTool for BashTool {
...
+fn redact_for_log(msg: &str, max_bytes: usize) -> String {
+    if msg.len() <= max_bytes {
+        return msg.to_string();
+    }
+    let mut cut = max_bytes;
+    while cut > 0 && !msg.is_char_boundary(cut) {
+        cut -= 1;
+    }
+    format!("{}…[truncated]", &msg[..cut])
+}
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/virtual_bash.rs]: Good catch — renamed to `truncate_for_log` in `d179a61`. The function never redacted (no masking, no pattern stripping), it only truncates with a visibility marker, so the new name matches the behavior.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/capabilities/virtual_bash.rs [hunk 2]
```diff
@@ -577,6 +579,71 @@ impl BackgroundExecutableTool for BashTool {
...
+/// truncated entries are visible in logs.
+fn redact_for_log(msg: &str, max_bytes: usize) -> String {
+    if msg.len() <= max_bytes {
+        return msg.to_string();
+    }
+    let mut cut = max_bytes;
+    while cut > 0 && !msg.is_char_boundary(cut) {
+        cut -= 1;
+    }
+    format!("{}…[truncated]", &msg[..cut])
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/capabilities/virtual_bash.rs]: Valid bug — the old implementation could overflow by the 14-byte `…[truncated]` marker. Fixed in `d179a61` by reserving marker bytes from the byte budget and falling back to a bare truncated slice when `max_bytes` is smaller than the marker. Added a regression test (`truncate_for_log_omits_marker_when_budget_is_too_small`) plus updated the existing budget test to assert `out.len() <= max_bytes`.

---
_Generated by [Claude Code](https://claude.ai/code)_
